nexus/types/src/external_api/params.rs

```diff
@@ -982,6 +982,21 @@ pub enum ExternalIpDetach {
     Floating { floating_ip: NameOrId },
 }
 
+/// A policy determining when an instance should be automatically restarted by
+/// the control plane.
+#[derive(Clone, Debug, Deserialize, Serialize, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub enum InstanceAutoRestart {
+    /// The instance should not be automatically restarted by the control plane
+    /// if it fails.
+    Never,
+    /// The control plane will make a best-effort attempt to ensure this
+    /// instance is running, but will not guarantee that the instance will
+    /// always be restarted. The control plane may choose not to restart this
```

Comment:
```suggestion
    /// If this instance is running and unexpectedly fails (e.g. due to a host
    /// software crash or unexpected host reboot), the control plane will make a
    /// best-effort attempt to restart it. The control plane may choose not to
    /// restart the instance to preserve the overall availability of the system.
```

A minor tweak to emphasize that the control plane only tries to make sure the instance remains running if it was running already.

Would it be worthwhile to start writing a guide in the docs repo explaining why instances can appear to reboot (akin to [this one](https://learn.microsoft.com/en-us/troubleshoot/azure/virtual-machines/windows/understand-vm-reboot) from Azure)? I guess if I thought to ask the answer is likely to be "yes." I can file an issue on the docs repo if you agree.

Reply:
Yeah, I think we should absolutely discuss this stuff in the docs --- I would love an issue to get that written down. We'll probably need to finish the implementation first though so that we're documenting the behavior as-built. :)